packages/jest-cli/src/TestNamePatternPrompt.js

```diff
@@ -30,108 +30,106 @@ const usage = () =>
 
 const usageRows = usage().split('\n').length;
 
-module.exports = (
-  config: Config,
-  pipe: stream$Writable | tty$WriteStream,
-  prompt: Prompt,
-) => {
-  class TestNamePatternPrompt {
-    // $FlowFixMe
-    _cachedTestResults;
-
-    constructor() {
-      (this: any).onChange = this.onChange.bind(this);
-    }
+module.exports = class TestNamePatternPrompt {
+  _cachedTestResults: Array<TestResult>;
+  _pipe: stream$Writable | tty$WriteStream;
+  _prompt: Prompt;
 
-    run(onSuccess: Function, onCancel: Function) {
-      pipe.write(ansiEscapes.cursorHide);
-      pipe.write(ansiEscapes.clearScreen);
-      pipe.write(usage());
-      pipe.write(ansiEscapes.cursorShow);
+  constructor(pipe: stream$Writable | tty$WriteStream, prompt: Prompt) {
+    this._pipe = pipe;
+    this._prompt = prompt;
+    (this: any).onChange = this.onChange.bind(this);
+  }
 
-      prompt.enter(this.onChange, onSuccess, onCancel);
-    }
+  run(onSuccess: Function, onCancel: Function) {
+    this._pipe.write(ansiEscapes.cursorHide);
+    this._pipe.write(ansiEscapes.clearScreen);
+    this._pipe.write(usage());
+    this._pipe.write(ansiEscapes.cursorShow);
 
-    onChange(pattern: string) {
-      pipe.write(ansiEscapes.eraseLine);
-      pipe.write(ansiEscapes.cursorLeft);
-      this.printTypeahead(pattern, 10);
-    }
+    this._prompt.enter(this.onChange, onSuccess, onCancel);
+  }
 
-    printTypeahead(pattern: string, max: number) {
-      const matchedTests = this.getMatchedTests(pattern);
+  onChange(pattern: string) {
+    this._pipe.write(ansiEscapes.eraseLine);
+    this._pipe.write(ansiEscapes.cursorLeft);
+    this.printTypeahead(pattern, 10);
+  }
 
-      const total = matchedTests.length;
-      const results = matchedTests.slice(0, max);
-      const inputText = `${chalk.dim(' pattern \u203A')} ${pattern}`;
+  printTypeahead(pattern: string, max: number) {
+    const matchedTests = this.getMatchedTests(pattern);
 
-      pipe.write(ansiEscapes.eraseDown);
-      pipe.write(inputText);
-      pipe.write(ansiEscapes.cursorSavePosition);
+    const total = matchedTests.length;
+    const results = matchedTests.slice(0, max);
+    const inputText = `${chalk.dim(' pattern \u203A')} ${pattern}`;
```

Comment:
Later we could extract this `\u203A` char to something readable, as it's shared across multiple files.